tests/integrationv2/test_npn.py

```diff
@@ -0,0 +1,251 @@
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", TLS_PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_s2n_client_npn(managed_process, cipher, curve, certificate, protocol, provider):
+    options = ProviderOptions(
+        port=next(available_ports),
+        cipher=cipher,
+        curve=curve,
+        key=certificate.key,
+        cert=certificate.cert,
+        protocol=protocol,
+        insecure=True,
+    )
+
+    client_options = copy.copy(options)
+    client_options.mode = Provider.ClientMode
+    # Flags to turn on NPN for s2nc
+    client_options.extra_flags = ['--alpn', PROTOCOL_LIST, '--npn']
+
+    server_options = copy.copy(options)
+    server_options.mode = Provider.ServerMode
+    # Flags to turn on NPN for OpenSSL server
+    server_options.extra_flags = ['-nextprotoneg', PROTOCOL_LIST]
+
+    server = managed_process(provider, server_options, timeout=5)
+    client = managed_process(S2N, client_options, timeout=5)
+
+    expected_protocol = 'http/1.1'
+
+    for results in server.get_results():
+        results.assert_success()
+        assert to_bytes(OPENSSL_SERVER_NPN_MARKER + expected_protocol) in results.stdout
+
+    for results in client.get_results():
+        results.assert_success()
+        assert to_bytes(S2N_NPN_MARKER) in results.stdout
+        assert to_bytes(S2N_APPLICATION_MARKER + expected_protocol) in results.stdout
+
+"""
+The s2n-tls client chooses a server-preferred protocol.
+"""
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", TLS_PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_s2n_client_npn_server_preference(managed_process, cipher, curve, certificate, protocol, provider):
+    options = ProviderOptions(
+        port=next(available_ports),
+        cipher=cipher,
+        curve=curve,
+        key=certificate.key,
+        cert=certificate.cert,
+        protocol=protocol,
+        insecure=True,
+    )
+
+    client_options = copy.copy(options)
+    client_options.mode = Provider.ClientMode
+    # Flags to turn on NPN for s2nc
+    client_options.extra_flags = ['--alpn', PROTOCOL_LIST, '--npn']
+
+    server_options = copy.copy(options)
+    server_options.mode = Provider.ServerMode
+    # Flags to turn on NPN for OpenSSL server
+    server_options.extra_flags = ['-nextprotoneg', PROTOCOL_LIST_ALT_ORDER]
+
+    server = managed_process(provider, server_options, timeout=5)
+    client = managed_process(S2N, client_options, timeout=5)
+
+    expected_protocol = 'h2'
```

Comment:
If this is exactly the same as test_s2n_client_npn and test_s2n_client_npn_no_overlap except for PROTOCOL_LIST_ALT_ORDER + expected_protocol, could you somehow combine these tests / let them reuse the common code?